### ISSUE TITLE
< iOS 10 support?

### ISSUE THREAD

**Issue by davidjb** [mssun/passforios#107]
Thank you for making and open-sourcing Pass for iOS! I'm the one who was struggling with https://github.com/davidjb/pass-ios and have since deprecated and pointed anyone finding those to your project.

With the current app, it requires iOS 10.2 or above.  Two devices I have are on iOS 9 and that's the end of the road for them (iPhone 4S).  If I adjust the deployment target back to (for example) `9.0`, the Swift compiler errors on a variety of things that were introduced in iOS 10, such as:

* passforios/passKit/Models/PasswordStore.swift:76:25: 'NSPersistentContainer' is only available on iOS application extension 10.0 or newer
* passforios/passKit/Models/PasswordStore.swift:80:50: 'NSPersistentStoreDescription' is only available on iOS application extension 10.0 or newer
* passforios/pass/AppDelegate.swift:105:35: 'NSPersistentContainer' is only available on iOS 10.0 or newer
* ... etc
  * For these, backported code for back to iOS7: https://gist.github.com/zwaldowski/83b8e0ca1f8c0ad8bece82cd1026c021 (appears to compile okay if I switch the existing code to using that)

* passforios/passExtension/ExtensionViewController.swift:113:9: 'UIImpactFeedbackGenerator' is only available on iOS application extension 10.0 or newer
* passforios/pass/Controllers/PasswordDetailTableViewController.swift:10:8: Module file's minimum deployment target is ios9.1 v9.1: /Users/david1/dev/passforios/Carthage/Build/iOS/FavIcon.framework/Modules/FavIcon.swiftmodule/arm.swiftmodule
* passforios/pass/Controllers/PasswordDetailTableViewController.swift:190:15: 'scheduledTimer(withTimeInterval:repeats:block:)' is only available on iOS 10.0 or newer
* passforios/pass/Controllers/PasswordDetailTableViewController.swift:419:30: 'open(_:options:completionHandler:)' is only available on iOS 10.0 or newer

In short, I'm wondering whether iOS 9 could be supported or if a PR to support it would be considered/accepted.  If you're thinking the latter, then I'd take suggestions on how you'd prefer to handle it before I go too far.  Thanks!

**Comment by mssun** [comment]
Sorry for the late reply. I am debating on this issue. Actually, I tried to port the project to target iOS 9 weeks ago. However, since there are many modifications on the code base, I gave up. Besides the issues on `NSPersistentContainer`, there are new APIs like `UIImpactFeedbackGenerator` used in the app. In addition, I was not sure how many users are still with iOS 9. Therefore, considering iOS 9 has been publicly jailbroken (which is dangerous for our app), I chose not to support to iOS 9  at that time.

We are very open to any PRs. If you think supporting iOS 9 is really a need and feel confident to make it happened, please go ahead and do it. We can target iOS 9.3, since it's the highest version which iPhone 4s can support. I guess you need to think about how to handle APIs which is not supported in iOS 9 (e.g., UIImpactFeedbackGenerator for generating haptic feedback).

At last, thanks for your donation and contributions, the most generous one ever :)

**Comment by davidjb** [comment]
Due to some changing circumstances, I've managed to get a device with iOS 10 so I'll close this off because as you say, the effort to backport was looking too great.  Thanks for your comments all the same.

You're very welcome about the donation -- thank you for making the project!

**Comment by eutampieri** [comment]
The haptic feedback could be feature gated. Is there any interest in this anymore?
